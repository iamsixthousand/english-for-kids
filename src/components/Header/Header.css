.Header {
  width: 100%;
  height: 90px;
  position: relative;
  display: flex;

  align-items: center;
  background-color: coral;
}

.SideBarOpenButton {
  padding: 0;
  margin: 0;
  transform: rotate(90deg);
  display: flex;
  align-items: center;
  justify-content: center;
  left: 10%;
  font-size: 4em;
  font-weight: 800;
  z-index: 5;
  cursor: pointer;
  transition: all 0.3s;
  position: absolute;
}

.SideBarOpenButton:hover {
  transition-delay: 0.1s;
  left: calc(10% - 10px);
  margin: 0;
  transform: rotate(270deg);
  color: white;
}

.ModeSelector {
  right: 10%;
  position: absolute;
}
