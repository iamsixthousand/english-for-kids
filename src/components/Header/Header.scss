@import '../../index.scss';
@import '../../styles.scss';

.Header {
  width: 100%;
  height: 90px;
  position: fixed;
  display: flex;
  align-items: center;
  z-index: 14;
  opacity: 0.9;
}

.SideBarOpenButton {
  padding: 0;
  margin: 0;
  transform: rotate(90deg);
  display: flex;
  align-items: center;
  justify-content: center;
  left: 10%;
  font-size: 3.5em;
  font-weight: 800;
  z-index: 5;
  cursor: pointer;
  transition: all 0.1s;
  position: absolute;
  width: 85px;
  height: 85px;
  user-select: none;
  &:hover {
    transition-delay: $t-delay;
    left: calc(10% + 5px);
    margin: 0;
    color: $secondary;
  }
}

.ModeSelector {
  right: 10%;
  position: absolute;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  z-index: 14;
}

.ModeText {
  margin-right: 5px;
  user-select: none;
  font-size: 15px;
  font-weight: $medium;
  font-family: $appfont;
}

@media (max-width: $minimum-screen) {
  .Header {
    width: $minimum-size;
  }
}
