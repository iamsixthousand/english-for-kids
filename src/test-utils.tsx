import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import gameProcessReducer from './state/gameProcessReducer';

const WithStoreProvider: React.FC = ({ children }) => {
  const myStore = createStore(
    combineReducers({
      gameProcess: gameProcessReducer,
    })
  );
  return <Provider store={myStore}>{children}</Provider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: WithStoreProvider, ...options });

export * from '@testing-library/react';
export { customRender as render };
