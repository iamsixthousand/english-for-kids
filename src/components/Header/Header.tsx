import React from 'react';
import { HeaderProps } from '../../interfaces/interfaces';
import './Header.css';

export const Header: React.FC<HeaderProps> = ({ sideBarToggle, setModeToggle }) => {
  return (
    <div className="Header">
      <div role="button" className="SideBarOpenButton" onClick={sideBarToggle} aria-hidden="true">
        |||
      </div>
      <div role="button" className="ModeSelector" onClick={setModeToggle} aria-hidden="true">
        MODE
      </div>
    </div>
  );
};
