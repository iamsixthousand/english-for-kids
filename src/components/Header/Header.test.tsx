/* eslint-disable no-unused-expressions */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Header } from './Header';
import { render } from '../../test-utils';

describe('CardHolder component:', () => {
  const mockSideBarToggle = jest.fn();
  const mockSetModeToggle = jest.fn();
  const mockSetIsBlockingToggle = jest.fn();

  it('matches snapshot', () => {
    const headerComp = render(
      <MemoryRouter>
        <Header
          sideBarToggle={mockSideBarToggle}
          setModeToggle={mockSetModeToggle}
          setIsBlockingToggle={mockSetIsBlockingToggle}
        />
      </MemoryRouter>
    );
    expect(headerComp).toMatchSnapshot();
  });
});
