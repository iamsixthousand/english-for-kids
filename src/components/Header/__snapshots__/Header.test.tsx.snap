// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHolder component: matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Header"
        style="background-image: url(/background.jpg);"
      >
        <div
          aria-hidden="true"
          class="SideBarOpenButton"
          data-testid="SideBarOpenButton"
          role="button"
        />
        <div
          class="ModeSelector"
        >
          <span
            class="ModeText"
          >
            TRAIN
          </span>
          <div
            class="Switch"
            style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 10px; transition: opacity 0.25s; user-select: none;"
          >
            <div
              class="react-switch-bg"
              style="height: 20px; width: 48px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 10px; cursor: pointer; transition: background 0.25s;"
            />
            <div
              class="react-switch-handle"
              style="height: 18px; width: 18px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.2); border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
            />
            <input
              aria-checked="false"
              data-testid="SwitchElement"
              role="switch"
              style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
              type="checkbox"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Header"
      style="background-image: url(/background.jpg);"
    >
      <div
        aria-hidden="true"
        class="SideBarOpenButton"
        data-testid="SideBarOpenButton"
        role="button"
      />
      <div
        class="ModeSelector"
      >
        <span
          class="ModeText"
        >
          TRAIN
        </span>
        <div
          class="Switch"
          style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 10px; transition: opacity 0.25s; user-select: none;"
        >
          <div
            class="react-switch-bg"
            style="height: 20px; width: 48px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 10px; cursor: pointer; transition: background 0.25s;"
          />
          <div
            class="react-switch-handle"
            style="height: 18px; width: 18px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.2); border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
          />
          <input
            aria-checked="false"
            data-testid="SwitchElement"
            role="switch"
            style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
